openforcefield/utils/toolkits.py

```diff
@@ -1784,17 +1799,16 @@ def to_file_obj(self, molecule, file_obj, file_format):
 
         """
         from rdkit import Chem
+
         file_format = file_format.upper()
         rdmol = self.to_rdkit(molecule)
-        rdkit_writers = {
-            'SDF': Chem.SDWriter,
-            'PDB': Chem.PDBWriter,
-            'SMI': Chem.SmilesWriter,
-            'TDT': Chem.TDTWriter
-        }
-        writer = rdkit_writers[file_format](file_obj)
-        writer.write(rdmol)
-        writer.close()
+        try:
+            writer = eval(self._toolkit_file_write_formats[file_format])(file_obj)
```

Comment:
Here we call `eval` on the string representation of the file writer class/function.